ntbs-service/Models/Entities/SocialContextBase.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using ExpressiveAnnotations.Attributes;
+using ntbs_service.Helpers;
+using ntbs_service.Models.Validations;
+
+namespace ntbs_service.Models.Entities
+{

```

Comment:
SocialContextVenue also contains address (the address of the venue, rather than previous addresses of the patient) with basically identical validation requirements, hence this base class. Postcode is a required feel for Address, but not Venue - hence it's defined separately in each model (as setting the attribute in each seemed fiddly from a quick google search - but obviously I could do that if you think that's better! The Regex and displayname are the same so we would avoid that repetition)

Reply:
There might be a quick solution using `RequiredIf("PostcodeRequired")` and letting each submodel implement a `PostcodeRequired` getter?
If that doesn't fire then that repetition seems fine!

Reply:
yes that worked, good idea!